dataset/put-metadata.go

```diff
@@ -0,0 +1,95 @@
+package dataset
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+
+	dphandlers "github.com/ONSdigital/dp-net/handlers"
+	"github.com/ONSdigital/dp-publishing-dataset-controller/model"
+	"github.com/ONSdigital/log.go/log"
+	"github.com/gorilla/mux"
+)
+
+// PutMetadata updates dataset, version and dimension data
+func PutMetadata(dc DatasetClient, zc ZebedeeClient) http.HandlerFunc {
+	return dphandlers.ControllerHandler(func(w http.ResponseWriter, r *http.Request, lang, collectionID, accessToken string) {
+		putMetadata(w, r, dc, zc, accessToken, collectionID, lang)
+	})
+}
+
+func putMetadata(w http.ResponseWriter, req *http.Request, dc DatasetClient, zc ZebedeeClient, userAccessToken, collectionID, lang string) {
+	ctx := req.Context()
+
+	if userAccessToken == "" {
```

Comment:
It looks like this is the same logic for the `getMetaData` handler function; could it be worthwhile extracting these two checks into a helper function(s)?